base/datatype_reference.bal

```diff
@@ -78,3 +78,68 @@ public type Reference record {|
     Identifier identifier?;
     string display?;
 |};
+
+# Create a relative Reference to FHIR Resource.
+#
+# + referenceResourceModel - FHIR Resource model of the Reference
+# + relativePathToReference - Relative path to Reference
+# + return - FHIRError if the Reference resource type is not defined, else FHIR Reference
+public function createRelativeFhirReference(typedesc<anydata> referenceResourceModel, string relativePathToReference) returns Reference|FHIRError {
+    ResourceDefinitionRecord? def = referenceResourceModel.@ResourceDefinition;
+    string? resourceType = def?.resourceType;
+
+    if resourceType is string {
+        return {
+            reference: string `${resourceType}/${relativePathToReference}`,
+            'type: resourceType
+        };
+    }
+    return createFHIRError("Resource type of the Reference resource model is not defined", ERROR, PROCESSING, errorType = PROCESSING_ERROR);
+}
+
+# Create an absolute Reference to FHIR Resource.
+#
+# + referenceResourceModel - FHIR Resource model of the Reference
+# + absolutePathToReference - Absolute path to Reference
+# + return - FHIRError if the Reference resource type is not defined, else FHIR Reference
+public function createAbsoluteFhirReference(typedesc<anydata> referenceResourceModel, string absolutePathToReference) returns Reference|FHIRError {
```

Comment:
Both `createRelativeFhirReference` and `createAbsoluteFhirReference` have same function body except for `reference` field in the returned record. Can be refactored to reuse the code.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/f0462073be05a0b5779b465f684236782a19c8f6

Reply:
I still this repeated code in both functions. 
https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/pull/20/files#diff-54423f438aa89d43e05dc851e636a3b58a97bf0ecfec0c9b9e5680f3097fc4d3R88-R93

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/65644c31189a8f8b2c141d3c131d5f802b219299